hack/make/.integration-daemon-start

```diff
@@ -74,6 +74,9 @@ if [ -f "/sys/fs/cgroup/cgroup.controllers" ]; then
 	fi
 fi
 
+tmp_dir=$(mktemp -d)
+trap 'rm -rf "$tmp_dir"' EXIT
```

Comment:
Given how sprawling and intertwined these scripts are, we do need to be really careful about things like `trap` (so we don't clobber our own traps), but I think we're currently fine.

(This file used to have a `trap` that invoked `integration-daemon-stop` but that was removed back in 4308e881cc00141a1dd187120d1c4e2106148282)

Reply:
Yeah, I noticed that -- the heavy use of `source` does make this hard to reason about. Still, the worst consequences are leaking temporary directories with a socket, so I think it's benign if someone breaks it temporarily.